test/e2e/gameserver_test.go

```diff
@@ -67,3 +52,87 @@ func TestCreateConnect(t *testing.T) {
 
 	assert.Equal(t, reply, "ACK: Hello World !\n")
 }
+
+func TestSDKSetLabel(t *testing.T) {
+	t.Parallel()
+	gs := defaultGameServer()
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(defaultNs, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
+	}
+
+	assert.Equal(t, readyGs.Status.State, v1alpha1.Ready)
+	reply, err := e2eframework.PingGameServer("LABEL", fmt.Sprintf("%s:%d", readyGs.Status.Address,
+		readyGs.Status.Ports[0].Port))
+
+	if err != nil {
+		t.Fatalf("Could ping GameServer: %v", err)
+	}
+
+	assert.Equal(t, "ACK: LABEL\n", reply)
+
+	// the label is set in a queue, so it may take a moment
+	err = wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
+		gs, err = framework.AgonesClient.StableV1alpha1().GameServers(defaultNs).Get(readyGs.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
+		}
+		return gs.ObjectMeta.Labels != nil, nil
+	})
+
+	assert.Nil(t, err)
+	assert.NotEmpty(t, gs.ObjectMeta.Labels["stable.agones.dev/sdk-timestamp"])
+}
+
+func TestSDKSetAnnotation(t *testing.T) {
+	t.Parallel()
+	gs := defaultGameServer()
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(defaultNs, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
+	}
+
+	assert.Equal(t, readyGs.Status.State, v1alpha1.Ready)
+	reply, err := e2eframework.PingGameServer("ANNOTATION", fmt.Sprintf("%s:%d", readyGs.Status.Address,
```

Comment:
This feels like we're going to need a better client for interacting with the gameserver at some point. If we could received and send a variable for instance we would be able to assert that the right value is stored.

However I think for now this would do the trick !